index.js

```diff
@@ -124,39 +126,68 @@ const headers = {
 				}));
 			}, []),
 
-			{ errorCount, warningCount } = report;
-
+			{
+				errorCount, warningCount
+			} = report;
 		return {
-			conclusion: errorCount > 0 ? 'failure' : 'success',
-			output: {
-				title: checkName,
-				summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
-				text: 'A little bit of text',
-				annotations
-			}
+			annotations,
+			errorCount,
+			warningCount
 		};
 	},
-	updateCheck = async (id, conclusion, output) =>
+	updateCheck = async (id, opts = {}) =>
 		await request(`https://api.github.com/repos/${ owner }/${ repo }/check-runs/${ id }`, {
 			method: 'PATCH',
 			headers,
 			body: {
 				name: checkName,
 				head_sha: checkSha,
-				status: 'completed',
-				completed_at: (new Date()).toISOString(),
-				conclusion,
-				output
+				...opts
 			}
 		}),
+	updateChecks = async (id, {
+		errorCount, warningCount, annotations: all
+	}) => {
+		const chunkSize = 50,
+			chunksLength = Math.ceil(all.length / 50);
+		await Promise.all(new Array(chunksLength).fill().map((_, i) => {
+			const annotations = all.slice(i * chunkSize, (i + 1) * chunkSize);
+			return updateCheck(id, i < chunksLength - 1
+				? {
+					status: 'in_progress',
+					output: {
+						title: checkName,
+						summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
+						annotations
+					}
+				}
+				: {
+					status: 'completed',
+					completed_at: (new Date()).toISOString(),
+					conclusion: errorCount > 0 ? 'failure' : 'success',
+					output: {
+						title: checkName,
+						summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
+						annotations
+					}
+				}
+			);
```

Comment:
I would put the ternary on it's own line.
```suggestion
			return updateCheck(
				id, 
				i < chunksLength - 1
					? {
						status: 'in_progress',
						output: {
							title: checkName,
							summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
							annotations
						}
					}
					: {
						status: 'completed',
						completed_at: (new Date()).toISOString(),
						conclusion: errorCount > 0 ? 'failure' : 'success',
						output: {
							title: checkName,
							summary: `${ errorCount } error(s), ${ warningCount } warning(s) found`,
							annotations
						}
					}
			);
```

Reply:
Updated this because I needed to handle the case when there are no annotations.